include/onnxruntime/core/optimizer/graph_transformer.h

```diff
@@ -32,11 +50,24 @@ class GraphTransformer {
     return desc_;
   }
 
+   /** Gets the level of this graph transformer. */
+  const TransformerLevel Level() const noexcept {
+    return level_;
+  }
+
+  /** Checks whether nodes assigned provider is compatible with this transformer. */
+  bool IsProviderCompatible(const std::string& providerType) const {
+    if (providerType.empty() || std::find(compatible_providers_.begin(), compatible_providers_.end(), providerType) != compatible_providers_.end()) {
```

Comment:
Would make more sense to me to take a const Node& given the method is to check if the Node's provider is compatible. Or if there's a real use case where a raw provider type needs to be checked, add a separate IsProviderCompatible method that takes a const Node&.

Reply:
Actually every transformer runs this check already : IsSupportedOptypeVersionAndDomain
I am planning to move execution provider check into this as well. Thoughts?

Reply:
If there are generic checks that every transformer is repeating, put them here. 

---
In reply to: [263110225](https://github.com/Microsoft/onnxruntime/pull/556#discussion_r263110225) [](ancestors = 263110225)